src/test/scala/extractors/ExtractorSpec.scala

```diff
@@ -0,0 +1,116 @@
+    Given("""domain string "test.com.ua" """)
+    val domain: String = "test.com.ua"
+
+    When("given string is unapplied using match (unapplySeq) and first element is taken as result")
+    val reversedHeadDomain: String = domain match {
+      case Domain(head, _*) => head
+    }
+
+    Then("""result should be value of top-level domain "ua" """)
+    reversedHeadDomain should equal("ua")
+  }
+
+  it should "return an empty Array for empty input string" in {
+    Given("empty domain string")
+    val domain: String = ""
+
+    When("given string is unapplied using match (unapplySeq) and first element is taken as result")
+    val reversedHeadDomain: String = domain match {
+      case Domain(head, _*) => head
+    }
+
+    Then("result should be equal to empty string")
+    reversedHeadDomain should equal("")
+  }
+
+  "Extractor.filterNotDomainsWithParticularHead function" should """take list of domains to input and return filtered list without domains with "in.ua" part""" in {
+    Given("""list of domains "test.in.ua", "test.com", "test.failed.in.ua", "test.passed.com", "in.ua" """)
+    val domainList: List[String] = List("test.in.ua", "test.com", "test.failed.in.ua", "test.passed.com", "in.ua")
+
+    When("this list is filtered using tested function")
+    val filteredList = Extractor.filterNotDomainsWithParticularHead(domainList)
+
+    Then("size of filtered list should be 2")
+    filteredList.size should equal(2)
+    And("""head element of filtered list should be "test.com" """)
+    filteredList.head should equal("test.com")
+    And(""" first element of filtered list should be "test.passed.com" """)
+    filteredList(1) should equal("test.passed.com")
+  }
+
+  it should """return empty list if there are no other domains except "in.ua" in input list""" in {
+    Given("""list of domains "test.in.ua", "test.failed.in.ua", "in.ua" """)
+    val domainList: List[String] = List("test.in.ua", "test.failed.in.ua", "in.ua")
+
+    When("this list is filtered using tested function")
+    val filteredList = Extractor.filterNotDomainsWithParticularHead(domainList)
+
+    Then("size of filtered list should be 0")
+    filteredList.size should equal(0)
+  }
+
+  "Extractor.filterEmailsWithParticularDomain function" should """take array of emails to input and print only emails with "gmail.com" domain (by default)""" in {
+    Given("""array of emails "test@gmail.com", "test@in.ua", "holy-scala@gmail.com", "scala@metagmail.com" """)
+    val emailArray: Array[String] = Array("test@gmail.com", "test@in.ua", "holy-scala@gmail.com", "scala@metagmail.com")
+    val printCheck = new ByteArrayOutputStream()
+
+    When("tested function is called for given array")
+    Console.withOut(printCheck) {
+      Extractor.filterEmailsWithParticularDomain(emailArray)
+    }
+    val consoleResult: String = printCheck.toString
+
+    Then("""it should print emails with "gmail.com" domain""")
+    consoleResult.contains("test@gmail.com") should equal(true)
```

Comment:
`should equal(true)` will produce unmeaningful messages if failed.

Reply:
Type of assertion has been changed to "should contain". Now, in case of fail we will have message like:
Array("test@gmail.com", "holy-scala@gmail.com") did not contain element "test2@gmail.com"